File: app/src/main/java/me/yangle/myphone/TableDisplayActivity.java
package me.yangle.myphone;

import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.ViewGroup;
import android.widget.TableRow;
import android.widget.TextView;

import org.json.JSONArray;
import org.json.JSONException;

class TableDisplayActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_table_display);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        try {
            JSONArray array = new JSONArray(getIntent().getStringExtra("json"));

            setTitle(array.getString(0));

            ViewGroup layout = (ViewGroup) findViewById(R.id.activity_table_display);
            for (int i = 1; i < array.length(); ++i) {
                JSONArray obj = array.getJSONArray(i);

                TableRow row = new TableRow(this);
                for (int j = 0; j < obj.length(); ++j) {
                    TextView column = new TextView(this);
                    column.setText(obj.getString(j));

                    row.addView(column);
                }

                layout.addView(row);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

    }
}

File: app/src/main/java/me/yangle/myphone/TableDisplayActivity.kt
package me.yangle.myphone

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.TableRow
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import org.json.JSONArray
import org.json.JSONException

class TableDisplayActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_table_display)
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
        try {
            val array = JSONArray(intent.getStringExtra("json"))
            title = array.getString(0)
            val layout = findViewById<View>(R.id.activity_table_display) as ViewGroup
            for (i in 1 until array.length()) {
                val obj = array.getJSONArray(i)
                val row = TableRow(this)
                for (j in 0 until obj.length()) {
                    val column = TextView(this)
                    column.text = obj.getString(j)
                    row.addView(column)
                }
                layout.addView(row)
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }
}